client/src/auth/slice.js

```diff
@@ -0,0 +1,64 @@
+import { createSlice } from '@reduxjs/toolkit';
+
+export const initialState = {
+  currentUser: {
+    id: '',
+    name: '',
+  },
+  loading: false,
+};
+
+const authSlice = createSlice({
+  name: 'auth',
+  initialState,
+  reducers: {
+    login: {
+      reducer: (state, action) => {
+        state.loading = true;
+        state.currentUser.id = action.payload.id;
+      },
+      prepare: (id, history) => {
```

Comment:
what does the reducer/prepare combo do as opposed to the other reducers in this map?